.github/workflows/serverless-integration.yml

```diff
@@ -11,7 +11,10 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-
+    strategy:
+      max-parallel: 2
```

Comment:
According to the GitHub docs for `max-parallel`:

> By default, GitHub will maximize the number of jobs run in parallel depending on the available runners on GitHub-hosted virtual machines.

So, I don't think you need this here.